proposals/NNNN-new-remote-control-modules-and-parameters.md

```diff
@@ -0,0 +1,512 @@
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Android / RPC ]
+
+## Introduction
+
+SDL remote control provides the ability for mobile applications to control certain components of the vehicle, such as, tune the radio, set the desired temperature or change the fan speed. However, currently only `CLIMATE` and `RADIO` modules are available for remote-control. Mobile application developers need more than basic climate or radio control. To meet developers' requirements, we propose to add three new modules (`AUDIO`, `LIGHT`, `HMI_SETTINGS`) and some new parameters (heatedXXXEnable and radio sisData) to the existing `CLIMMATE` and `RADIO` modules respectively. This proposal only defines new data types and new parameters added to the existing data types. It does not add or change any remote control RPCs.
+
+## Motivation
+
+Mobile application developers would like to develop applications that take fully advantage of the capability that the vehicle provides. There are many cases that mobile application developers ask for new functions of SDL remote control. For example,
+
+- Some mobile navigation applications would like to use the same distance unit and day/night display mode as the infotainment system does.
+
+- Some climate control applications want to control more elements in climate control, such as heated windshield, heated rear window, heated steering wheel and heated mirrors if the vehicle has them installed.
+
+- Some media applications want to use the existing radio/CD/DVD installed in vehicle, choose from the available audio sources, and switch audio source between in-vehicle radio and mobile application (internet radio) when necessary.
+
+- Some light control applications want to change the ambient lights of the vehicle and/or turn on/off certain exterior lights (while the vehicle is in parking).
+
+- Some radio control applications need more than the traditional RDS data, they need HD radio status data (especially the station id in the SIS data).
+
+
+## Proposed solution
+
+As usual, the remote control mobile applications use the following RPCs
+- `GetSystemCapability` with `SystemCapabilityType=REMOTE_CONTROL` to read available control modules and available control items within each module;
+- `GetInteriorVechileData` to read the current status value of a remote-control module;
+- `GetInteriorVechileData` with `subscribe`=true/false to register/un-register a module's data change notifications;
+- `SetInteriorVechileData` to change the settings/status of a remote-control module;
+- `OnInteriorVechileData` to receive any remote-control module value change notifications;
+
+In addition, in a GetInteriorVechileData response or OnInteriorVechileData notifications, an `xyzmoduleControlData` indicates the current value of the module status; in SetInteriorVechileData request, an `xyzmoduleControlData` indicates the required target state of the module.
+
+#### Add there new module types to the existing module list.
```

Comment:
Please change to "Add new module types to the existing module list."

Reply:
there -> three